src/python/graplctl/graplctl/queues/commands.py

```diff
@@ -1,61 +1,43 @@
 from __future__ import annotations
 
-import dataclasses
-from typing import TYPE_CHECKING
-
 import click
-from graplctl.common import GraplctlState, pass_graplctl_state
+from graplctl.common import State, pass_graplctl_state
 from graplctl.queues.lib import list_dlqs_for_deployment, redrive_from_dlq
 
-if TYPE_CHECKING:
-    from mypy_boto3_sqs import SQSClient
-
-
-@dataclasses.dataclass
-class QueuesObj:
-    sqs_client: SQSClient
-
 
-pass_queues_obj = click.make_pass_decorator(QueuesObj)
-
-
-@click.group(help="Manipulate work queues")
+@click.group()
 @click.pass_context
 @pass_graplctl_state
 def queues(
-    graplctl_state: GraplctlState,
+    graplctl_state: State,
     ctx: click.Context,
 ) -> None:
-    sqs_client = graplctl_state.boto3_session.client(
-        "sqs", region_name=graplctl_state.grapl_region
-    )
-    ctx.obj = QueuesObj(sqs_client)
+    """manipulate work queues"""
+    pass
 
 
-@queues.command(help="List redrivable queues")
-@pass_queues_obj
+@queues.command()
 @pass_graplctl_state
 def ls(
-    graplctl_state: GraplctlState,
-    queues_obj: QueuesObj,
+    graplctl_state: State,
 ) -> None:
-    queues = list_dlqs_for_deployment(graplctl_state, queues_obj.sqs_client)
+    """list redrivable queues"""
+    queues = list_dlqs_for_deployment(graplctl_state, graplctl_state.sqs)
     click.echo("\n".join(queues))
 
 
-@queues.command(help="Redrive messages from a dead letter queue")
-@pass_queues_obj
+@queues.command()
 @pass_graplctl_state
 @click.argument(
     "dlq_url",
 )
 def redrive(
-    graplctl_state: GraplctlState,
-    queues_obj: QueuesObj,
+    graplctl_state: State,
     dlq_url: str,
 ) -> None:
+    """redrive messages from a dead letter queue"""
```

Comment:
i am kinda curious about the moving the help= to the function; does it actually read the docstring?

Reply:
yes, it does:
```bash
(venv) jgrillo@penguin:~/src/grapl$ bin/graplctl -r us-east-1 -n jgrillo-test -g jgrillo-test -p lol aws --help
Usage: graplctl aws [OPTIONS] COMMAND [ARGS]...

  commands for managing grapl aws resources

Options:
  --help  Show this message and exit.

Commands:
  deploy     deploy grapl to aws
  destroy    tear down grapl in aws
  provision  provision the grapl deployment
```